### DISCUSSION TITLE
Using `umulExtended` in GLSL

### DISCUSSION THREAD

**Discussion by HailToDodongo** [gfx-rs/wgpu#4749] [Q&A] [answered]
Hi, 
I'm currently porting a project which uses GLSL compute-shaders.
It was originally using a library called "emu" which intern uses an old version of wgpu.
Since it's no longer maintained i tried using wgpu directly now.

The only issue i had so far was that the shader uses `umulExtended` which does not seem to be supported (i guess due to WGSL not supporting it).
So enabling the SPIR-V pass-through mode is the only option right now.

What is strange is that the unmaintained emu library, using an old wgpu version, did support this, seemingly without a pass-through.

Is there any way to use `umulExtended` right now with GLSL/SPIR-V shaders?

I only intend this to be usable as a native version anyways, and the main reason for a non pass-through for me would be to avoid MoltenVK on MacOS.

**Comment by teoxoy** [comment, marked as answer]
Hi, this is not currently possible. There was a time when `wgpu` was using `spirv-cross` for shader transpilation (which is most likely why `umulExtended` was previously supported) but we only use `naga` now.

Related: WGSL will probably get this (https://github.com/gpuweb/gpuweb/issues/1565) but it doesn't look like it's a priority right now.

**Reply by HailToDodongo** [reply]
Thanks for the answer, then it makes sense why it worked before.
I guess somehow using spriv-cross to compile the metal shader manually and feeding that into wgpu is not possible?

Otherwise i can live with MoltenVK for now and just wait until the support is officially in.

**Reply by teoxoy** [reply]
> I guess somehow using spriv-cross to compile the metal shader manually and feeding that into wgpu is not possible?

It isn't, as it's more maintenance work to support multiple transpilers.
